.github/workflows/pandora-ci.yml

```
@@ -25,28 +25,25 @@ concurrency:
 
 jobs:
   unit-tests:
-    permissions: write-all
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
 
     strategy:
       fail-fast: false
       max-parallel: 10
       matrix:
-        godot-version: ['4.0.4', '4.1.1', '4.2']
+        godot-version: ["4.0.4", "4.1.1", "4.2"]
 
     name: "🤖 CI on Godot ${{ matrix.godot-version }}"
-    uses: ./.github/workflows/unit-tests.yml
-    with:
-      godot-version: ${{ matrix.godot-version }}
-
-  finalize:
-    if: ${{ !cancelled() }}
-    runs-on: ubuntu-latest
-    name: Final Results
-    needs: [unit-tests]
     steps:
-      - run: exit 1
-        if: >-
-          ${{
-               contains(needs.*.result, 'failure')
-            || contains(needs.*.result, 'cancelled')
-          }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Run GDUnit4 tests
+        uses: MikeSchulze/gdUnit4-action@v1.1.1
+        with:
+          godot-version: ${{ matrix.godot-version }}
+          godot-status: "stable"
+          timeout: 3
+          paths: "res://test"
```

To keep the installed GdUnit4 when you have already installed
Using only the newest version when you build against minimum version of Godot 4.2.0
```suggestion
        with:
          godot-version: ${{ matrix.godot-version }}
          godot-status: "stable"
          version: "installed"
          timeout: 3
          paths: "res://test"
```